bundles/org.openhab.binding.solarforecast/src/main/java/org/openhab/binding/solarforecast/internal/actions/SolarForecastActions.java

```diff
@@ -0,0 +1,233 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.solarforecast.internal.actions;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+
+import javax.measure.MetricPrefix;
+import javax.measure.quantity.Energy;
+import javax.measure.quantity.Power;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.automation.annotation.ActionInput;
+import org.openhab.core.automation.annotation.RuleAction;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.binding.ThingActions;
+import org.openhab.core.thing.binding.ThingActionsScope;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * ActionsS to query forecast objects
+ *
+ * @author Bernd Weymann - Initial contribution
+ */
+@ThingActionsScope(name = "solarforecast")
+@NonNullByDefault
+public class SolarForecastActions implements ThingActions {
+    private final Logger logger = LoggerFactory.getLogger(SolarForecastActions.class);
+    private Optional<ThingHandler> thingHandler = Optional.empty();
+
+    @RuleAction(label = "@text/actionDayLabel", description = "@text/actionDayDesc")
+    public State getDay(
+            @ActionInput(name = "localDate", label = "@text/actionInputDayLabel", description = "@text/actionInputDayDesc") LocalDate localDate,
+            String... args) {
+        if (thingHandler.isPresent()) {
+            List<SolarForecast> l = ((SolarForecastProvider) thingHandler.get()).getSolarForecasts();
+            if (!l.isEmpty()) {
+                QuantityType<Energy> measure = QuantityType.valueOf(0, Units.KILOWATT_HOUR);
+                for (Iterator<SolarForecast> iterator = l.iterator(); iterator.hasNext();) {
+                    SolarForecast solarForecast = iterator.next();
+                    State s = solarForecast.getDay(localDate, args);
+                    if (s instanceof QuantityType<?>) {
+                        measure = measure.add((QuantityType<Energy>) s);
```

Comment:
corrected

Reply:
This doesn't work like proposed. Second cast to concrete QuantityType needed

```
if (s instanceof QuantityType<?> quantityState) {
                        measure = measure.add((QuantityType<Energy>)quantityState)
```

Reply:
Sorry, maybe this will work better?
```
                    QuantityType<Energy> s = solarForecast.getDay(localDate, args);
                    if (s instanceof QuantityType<Energy> quantityState) {
                        measure = measure.add(quantityState);
```

Reply:
Technically this works but I think this check is outdated.
In initial version `UNDEF` State was possible so `instanceof `check needed to be performed.
Now most problems are handled with exceptions and `UNDEF` isn't returned anymore.